test/models/IllnessStartDateSchema.test.ts

```diff
@@ -0,0 +1,157 @@
+        ]);
+    });
+
+    it('should reject blank values', () => {
+        const validationResult = validator.validate({
+            day: ' ',
+            month: ' ',
+            year: ' ',
+            illnessStart: ' '
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject invalid date patterns', () => {
+        const validationResult = validator.validate({
+            day: '001',
+            month: 'abc',
+            year: '*@&',
+            illnessStart: '0'
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+
+    it('should reject invalid (non-existing) date', () => {
+        const validationResult = validator.validate({
+            day: '33',
+            month: '01',
+            year: '2020',
+            illnessStart: '2020-01-33'
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+
+    it('should reject invalid (format) date', () => {
+        const validationResult = validator.validate({
+            day: '01',
+            month: '01',
+            year: '2020',
+            illnessStart: '2020-01-01'
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject invalid (future) date', () => {
+        const validationResult = validator.validate({
+            day: '01',
+            month: '01',
+            year: '2030',
+            illnessStart: new Date('2030-01-01')
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(startDateField, 'Start date must be today or in the past')
+        ]);
+    });
+
+    it('should accept valid date values', () => {
+        const validationResult = validator.validate({
+            day: '01',
+            month: '01',
+            year: '2020',
+            illnessStart: new Date('2020-01-01')
+        });
+        assertValidationErrors(validationResult, []);
+    });
+    it('should accept single digit date values', () => {
+        const validationResult = validator.validate({
+            day: '1',
+            month: '1',
+            year: '2020',
+            illnessStart: new Date('2020-01-01')
```

Comment:
This date validation feels so artificial due to the fact both date ingredients and full date is validated separately while values should be in sync. Crazy. As a result we have a test that tests single digits values but ISO date is correct ISO date. I understand validator will prepare data for validation but looking at that tests gives me mixed feelings :(